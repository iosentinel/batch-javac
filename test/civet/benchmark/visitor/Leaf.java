package benchmark.visitor;

public class Leaf extends Tree {
	int val;

	Leaf(int x) {
		this.val = x;
	}

	int accept(TreeVisitor t) {
		return t.visitLeaf(this);
	}

	int getVal() {
		return val;
	}
}
