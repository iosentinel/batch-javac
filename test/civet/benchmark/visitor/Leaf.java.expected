package benchmark.visitor;

public class Leaf extends Tree {
    int val;
    
    Leaf(int x) {
        super();
        this.val = x;
    }
    
    int accept(TreeVisitor t) {
        return t.visitLeaf(this);
    }
    
    int getVal() {
        return val;
    }
    
    int accept$1002() {
        return benchmark.visitor.FoldBinOp.visitLeaf$1003(this);
    }
    
    int accept$1007() {
        return benchmark.visitor.FoldBinOp.visitLeaf$1008(this);
    }
    
    int accept$1012() {
        return benchmark.visitor.CountOcc.visitLeaf$1013(this);
    }
}